lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -608,6 +651,24 @@ def code_to_exec(code=code, module=module, ctx=ctx, rerun_data=rerun_data):
             # which reads self._requests and checks for rerun data
             self._session_state[SCRIPT_RUN_WITHOUT_ERRORS_KEY] = run_without_errors
 
+            is_single_page_app = len(self._pages_manager.get_pages()) == 0
+            if (
+                is_single_page_app
+                and not ctx.pages_manager.is_mpa_v1
```

Comment:
the reason we don't show this in case of MPAv1 is because we automatically fallback to the main script there, is that correct?